lib/concurrent/mvar.rb

```diff
@@ -68,10 +68,14 @@ def take(timeout = nil)
 
         # If we timed out we'll still be empty
         if unlocked_full?
-          value = @value
-          @value = EMPTY
-          @empty_condition.signal
-          apply_deref_options(value)
+          if block_given?
+            yield @value
```

Comment:
This doesn't seems to make sense to me. Why would take remove the value in this case?